src/MultilineStrings.jl

```diff
@@ -0,0 +1,253 @@
+        :folded
+    elseif style_char == 'l' || style_char == '|'
+        :literal
+    elseif style_char == '\0'
+        DEFAULT_STYLE
+    else
+        throw(ArgumentError("Unknown style indicator: $(repr(style_char))"))
+    end
+
+    chomp = if chomp_char == 'c' || yaml_chomp
+        :clip
+    elseif chomp_char == 's' || chomp_char == '-'
+        :strip
+    elseif chomp_char == 'k' || chomp_char == '+'
+        :keep
+    elseif chomp_char == '\0'
+        DEFAULT_CHOMP
+    else
+        throw(ArgumentError("Unknown chomping indicator: $(repr(chomp_char))"))
+    end
+
+    return multiline(str, style, chomp)
+end
+
+function multiline(str::AbstractString, style::Symbol, chomp::Symbol)
+    # Append an additional, non-space, character to force one more iteration of the style
+    # loop
+    str *= ETX
+
+    out = IOBuffer()
+    num_newlines = 0  # The number of newlines at the end of the string
+    prev = curr = '\0'
+
+    # Replace newlines with spaces (folded)
+    if style === :folded
+        # The code below is equivalent to these two regexes:
+        # ```
+        # str = replace(str, r"(?<=\S)\n(?=\S)" => " ")
+        # str = replace(str, r"(?<=\n)\n(?=\S)" => "")
+        # ```
+
+        for next in str
+            if curr == '\n'
+                if !isspace(next) && next != ETX
+                    if prev == '\n'
+                        # Skip last newline in a sequence of sequential blank lines
+                    elseif !isspace(prev)
+                        # Replace a single newline with a space
+                        write(out, ' ')
+                    else
+                        num_newlines += 1
+                    end
+                else
+                    num_newlines += 1
+                end
+            elseif curr != '\0'
+                # Insert newlines which were determined to not be at the end of the string
+                if num_newlines > 0
+                    write(out, "\n" ^ num_newlines)
+                    num_newlines = 0
+                end
+
+                write(out, curr)
+            end
+
+            prev = curr
+            curr = next
+        end
+
+    # Keep newlines (literal)
+    elseif style === :literal
+        for next in str
+            if curr == '\n'
+                num_newlines += 1
+            elseif curr != '\0'
+                # Insert newlines which were determined to not be at the end of the string
+                if num_newlines > 0
+                    write(out, "\n" ^ num_newlines)
+                    num_newlines = 0
+                end
+
+                write(out, curr)
+            end
+
+            curr = next
+        end
+    else
+        throw(ArgumentError("Unknown style indicator: $style"))
+    end
+
+    # Single newline at end (clip)
+    if chomp === :clip
+        num_newlines > 0 && write(out, '\n')
+
+    # No newline at end (strip)
+    elseif chomp === :strip
```

Comment:
Not necessary?

Reply:
That is correct. Mainly here for consistency. I think it's clearer having this explicitly stated